### PR TITLE
Make Custom IScalarPriceOracle to include resolutionDate param

### PR DESCRIPTION
**Pull request by emaG3m**

### CODE REVIEW

File: contracts/Futarchy.sol [hunk 1]
```diff
@@ -119,7 +124,7 @@ contract Futarchy is AragonApp, IForwarder {
 
       FutarchyOracle futarchyOracle = futarchyOracleFactory.createFutarchyOracle(
         ERC20Gnosis(token),
-        priceOracleFactory.createCentralizedOracle("QmWmyoMoctfbAaiEs2G46gpeUmhqFRDW6KWo64y5r581Vz"),
+        Oracle(priceOracleFactory.createOracle("QmWmyoMoctfbAaiEs2G46gpeUmhqFRDW6KWo64y5r581Vz", priceResolutionDate)),
```
Comment (emaG3m) [inline contracts/Futarchy.sol]: I wonder if we can do way with an `ipfsHash` altogether...

Comment (mikec) [reply contracts/Futarchy.sol]: @emaG3m I think the Gnosis "CentralizedOracle" expects the hash ... we can just use our own that doesn't expect a hash

File: contracts/Futarchy.sol [hunk 2]
```diff
@@ -62,22 +65,22 @@ contract Futarchy is AragonApp, IForwarder {
...
     * @param _priceOracleFactory oracle factory used to create oracles that resolve price after all trading is closed
     * @param _lmsrMarketMaker market maker library that calculates prediction market outomce token prices
     * FutarchyOracleFactory comes from Gnosis https://github.com/gnosis/pm-contracts/blob/v1.0.0/contracts/Oracles/FutarchyOracleFactory.sol
-    * TODO: BoundsOracle - Calculates the upper and lower bounds for creating a new FutarchyOracle
-    * TODO: PriceFeedOracle - Is used to resolve all futarchy decision markets
     **/
     function initialize(
       uint24 _fee,
       uint _tradingPeriod,
+      uint _timeToPriceResolution,
```
Comment (emaG3m) [inline contracts/Futarchy.sol]: I was airing on the side of consistency with gnosis contracts with naming `tradingPeriod`. If we want to changes these to something like `categoricalTradingPeriod` `scalarTradingPeriod` perhaps we can make that a possibility

File: contracts/Futarchy.sol [hunk 3]
```diff
@@ -134,7 +139,7 @@ contract Futarchy is AragonApp, IForwarder {
       require(token.approve(futarchyOracle, marketFundAmount));
       futarchyOracle.fund(marketFundAmount);
 
-      emit StartDecision(decisionId, msg.sender, metadata, decisions[decisionId].futarchyOracle, lowerBound, upperBound);
+      emit StartDecision(decisionId, msg.sender, metadata, decisions[decisionId].futarchyOracle, lowerBound, upperBound, priceResolutionDate);
```
Comment (mikec) [inline contracts/Futarchy.sol]: @emaG3m can you add `startDate` and `tradingPeriod` to this event?

File: contracts/Oracles/CentralizedTimedOracle.sol [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+  address public owner;
+  bytes public ipfsHash;
+  bool public isSet;
+  int public outcome;
+
+  modifier isOwner () {
+    // Only owner is allowed to proceed
+    require(msg.sender == owner);
+    _;
+  }
```
Comment (mikec) [inline contracts/Oracles/CentralizedTimedOracle.sol]: might want to use Zeppelin Ownable

**Pull request merged**